atc/exec/step_metadata.go

```diff
@@ -62,5 +64,15 @@ func (metadata StepMetadata) Env() []string {
 		env = append(env, "ATC_EXTERNAL_URL="+metadata.ExternalURL)
 	}
 
+	if metadata.CreatedBy != nil {
```

Comment:
Another solution could be the ability to mark a job (or step, but probably more appropriate for job) as `auditable: true`. This would make it so that the userinfo is passed along to every step, along with outputting a log entry with the userinfo every time a build is created.

Heck, maybe in the future we can expand on the audit ideas by instrumenting in OPA requests when certain builds are created.